.github/workflows/hugo.yml

```diff
@@ -0,0 +1,37 @@
+name: Generate and upload Hugo docs
+
+on:
+  push:
+    branches: master
+
+jobs:
+  ocaml:
+    name: Docs
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.119.0'
+
+      - name: Build
+        run: |
+          cd doc
+          hugo --minify
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.ACTIONS_DOCS_DEPLOY_KEY }}
```

Comment:
I think this should be part of the docs.yml workflow so it's easy to see both are deploying to the same repo.

Reply:
I wasn't sure what the best practice is... But they aren't actually deploying to the same repo. The other one goes to xapi-storage. And the API docs are pulled by xapi-project.github.io.

Reply:
Perhaps we'll merge all that at some point.

Reply:
Talked to @psafont... we'll merge all docs-related Actions later.